### DISCUSSION TITLE
Organizing decks

### DISCUSSION THREAD

**Discussion by Julian702** [kioku-project/kioku#170] [Ideas]
# Concept 1
Decks in a group that the user is a member of are not included in the learned decks by default. They are not taken into account by the SRS algorithm. This is necessary because groups can also contain decks that are not relevant for every user. To prevent this, an unnecessarily large number of groups with similar user groups would have to be created. 
This default includes the home group for consistency reasons, which also allows decks to be archived. 

### Continue-learning list
The purpose of this list is to contain all decks that the user is learning. All decks on the list are taken into account by the SRS algorithm.
Decks are automatically added as soon as the user starts learning them. Removing decks from this list when they are no longer of interest is left to the user. 

### Bookmark list
This list can be manually customized by the user and can contain all publicly available decks. Only the user can add or remove decks. For this reason, it is possible for a deck to be both bookmarked and on the continue learning list.

## Technical details
These two lists would each be represented by a new relation in the database between user and deck. This also allows us to store the algorithm for each learned deck during the evaluation of different learning algorithms.

# Concept 2
Each deck is assigned exactly one state for each user. A deck can not have more than one state. This could be one of the following:

### Default:
These are all decks in groups in which the user is a member, including the home group. 

### Bookmark:
These are decks saved by the user for the future.

### Edit:
These are the decks the user is currently editing. 
This status is not necessarily needed.

### Active:
These are the decks that the user is learning. Just these decks are included in the SRS algorithm.

### Archived: 
These are the decks that have already been learned or are no longer relevant.
This status is not necessarily needed, but could be added to display the already learned decks in a special way, with e.g. more statistics. 

Moving the decks between the different states is left to the user which would be realized via a context menu. 
When learning a deck for the first time, the current status of the deck will be changed to Active .

**Comment by Julian702** [comment]
@TomRomeo and I have been thinking about how users should be able to organize their decks. This is the result of our discussion. If anything is unclear, please feel free to ask and add your suggestions.

**Comment by Julian702** [comment]
After a discussion with @JJthere , there is now a second proposed solution. Not both of us prefer this alternative.

**Reply by leonie2003** [reply]
What's the difference between default and archived in the second solution. Aren't archived decks also not taken into account by the SRS algorithm. Couldn't you therefore just go with three options?

**Reply by Julian702** [reply]
Yes, I absolutely agree!

**Comment by JJthere** [comment]
## Concept 1

### Advantages: 
- Bookmarklist is more customizable, also allowing the user to save already learned decks 

### Disadvantages:
- Removing from the Continue-Learning needs to be displayed somewhere else than the bookmark-icon and therefore leads to more lost space on the deck or has to get unintuitive 

## Concept 2

### Advantages: 
- Easily adjustable amount of status 
- Intuitive, because every status displays precisely what the user is currently doing with the deck

### Disadvantages:
- Contextmenu might get unintuitive 

## Conclusion:
To really decide which concept is better we need to discuss what the bookmark function is supposed to do.  
In my opinion it should just be to save decks the user wants to learn in the future.
